contracts/bonding/BondingManager.sol

```diff
@@ -860,21 +833,28 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
         totalBonded = totalBonded.add(_rewards);
     }
 
-    /*
-     * @dev Update a delegator with token pools shares from its lastClaimTokenPoolsSharesRound through a given round
+    /**
+     * @dev Update a delegator with token pools shares from its lastClaimRound through a given round
      * @param _delegator Delegator address
      * @param _endRound The last round for which to update a delegator's stake with token pools shares
      */
-    function updateDelegatorWithTokenPoolsShares(address _delegator, uint256 _endRound) internal {
+    function updateDelegatorWithEarnings(address _delegator, uint256 _endRound) internal {
         Delegator storage del = delegators[_delegator];
 
         // Only will have earnings to claim if you have a delegate
         // If not delegated, skip the earnings claim process
         if (del.delegateAddress != address(0)) {
+            // Cannot claim earnings for more than 20 rounds
+            // This is an arbitrary number to cause transactions to fail early if
+            // we know they will require too much gas to loop through all the necessary rounds to claim earnings
+            // The user should instead manually invoke `claimEarnings` to split up the claiming process
+            // across multiple transactions
+            require(_endRound.sub(del.lastClaimRound) <= 20);
```

Comment:
So this number is pretty arbitrary and it might be debatable whether this early failure is *that* useful - the idea is that if the number of rounds to claim through is known to be too big, `claimEarnings` and any bonding function with the `autoClaimEarnings` modifier will fail early instead of trying to loop through all the rounds thereby consuming all the necessary gas. This can always be determined off-chain, but this check just adds an extra safeguard in case anyone does call a function that tries to claim earnings through too many rounds